modules/engine/engine.cpp

```diff
@@ -455,7 +455,12 @@ static cell AMX_NATIVE_CALL set_view(AMX *amx, cell *params) {
 			g_pFunctionTable_Post->pfnPlayerPostThink=PlayerPostThink_Post;
 
 			plinfo[iIndex].iViewType = CAMERA_3RDPERSON;
-			pNewCamera = CREATE_NAMED_ENTITY(MAKE_STRING("info_target"));
+
+			if (!(pNewCamera = CREATE_NAMED_ENTITY(MAKE_STRING("info_target"))))
```

Comment:
I really like these kind of assignments since they save some unnecessary lines of code, but since it's too easy to confuse visually assignment and comparison I'm generally against using them and just assume every conditional with an assignment is a bug. I think some compilers or static analyzers even emit warnings unconditionally about them.